main.go

```diff
@@ -25,65 +25,80 @@ import (
+	RenewDeadline        time.Duration `long:"leader-elect-renew-deadline" default:"10s" description:"Interval between attempts by the acting master to renew a leadership slot before it stops leading (duration string)." env:"IPAM_RENEW_DEADLINE"`
+	RetryPeriod          time.Duration `long:"leader-elect-retry-period" default:"2s" description:"Duration the clients should wait between attempting acquisition and renewal of a leadership (duration string)." env:"IPAM_RESOURCE_LOCK"`
+	ResourceLock         string        `long:"leader-elect-resource-lock" default:"leases" description:"The type of resource object that is used for locking. Supported options are 'leases', 'endpoints', 'configmaps'." env:"IPAM_RESOURCE_LOCK_NAME"`
+	ResourceName         string        `long:"leader-elect-resource-name" default:"node-ipam-controller" description:"The name of the resource object that is used for locking." env:"IPAM_RESOURCE_NAME"`
+}
 
-	flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
-	flag.StringVar(&apiServerURL, "apiserver", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
-	flag.StringVar(&healthProbeAddr, "health-probe-address", ":8081", "Specifies the TCP address for the health server to listen on.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", true, "Enable leader election for the controller manager. Ensures there is only one active controller manager.")
-	flag.DurationVar(&leaseDuration, "leader-elect-lease-duration", 15*time.Second, "Duration that non-leader candidates will wait to force acquire leadership (duration string).")
-	flag.DurationVar(&renewDeadline, "leader-elect-renew-deadline", 10*time.Second, "Interval between attempts by the acting master to renew a leadership slot before it stops leading (duration string).")
-	flag.DurationVar(&retryPeriod, "leader-elect-retry-period", 2*time.Second, "Duration the clients should wait between attempting acquisition and renewal of a leadership (duration string).")
-	flag.StringVar(&resourceLock, "leader-elect-resource-lock", "leases", "The type of resource object that is used for locking. Supported options are 'leases', 'endpoints', 'configmaps'.")
-	flag.StringVar(&resourceName, "leader-elect-resource-name", "node-ipam-controller", "The name of the resource object that is used for locking.")
+func (c *config) load() error {
+	// allows using true/false in the parameters
+	_, err := flags.NewParser(c, flags.Default|flags.AllowBoolValues).ParseArgs(os.Args)
+	if err != nil {
+		return err
+	}
+	return nil
+}
 
+func main() {
 	c := logsapi.NewLoggingConfiguration()
 	logsapi.AddGoFlags(c, flag.CommandLine)
-	flag.Parse()
+
+	conf := config{EnableLeaderElection: true}
```

Comment:
not familiar with this framework, does the library sets this to false if the flag is set and ignores it if the flag is not set @mneverov

Reply:
We [discussed](https://github.com/kubernetes-sigs/node-ipam-controller/pull/20#discussion_r1694767260) that by default the controller should run in leader election mode. The `go-flags` library does not have default values for boolean flags and suggests to use reverted flags like `--disable-leader-election` which will be on by default and can be overwritten.
To workaround this and provide the same flag semantics as in #20 @Peac36 added the line above such that if `--enable-leader-election` flag is not provided the controller will still run in leader election mode (again as before).
To disable leader election one can use `--enable-leader-election=false`.

Reply:
Make sense 

/lgtm

Reply:
So the new library differentiate between flag not set and set to false

Reply:
When `--enable-leader-election` or `--enable-leader-election=true` are provided as well as when the flag is not provided at all the controller will start in leader election mode.
Disabling leader election is possible via `--enable-leader-election=false`